File: frontend/src/app/manage/model/ManageDto.ts
import {Dto} from '../../common/Dto';
import {Color} from '../../core/model/Color';

export interface ManageDto {
    institutions: Array<InstitutionDto>
}

export interface InstitutionDto extends Dto {
    id: number,
    name: string,
    schoolClasses: Array<SchoolClassDto>
}

export interface SchoolClassDto extends Dto {
    id: number,
    institutionId: number,
    normalizedName: string,
    schoolYears: Array<SchoolYearDto>
}

export interface SchoolYearDto extends Dto {
    id: number,
    schoolClassId: number,
    name: string,
    validFrom: Date,
    validTo: Date,
    semesters: Array<SemesterDto>
}

export interface SemesterDto extends Dto {
    id: number,
    schoolYearId: number,
    name: string,
    validFrom: Date,
    validTo: Date,
    subjects: Array<SubjectDto>
}

export interface SubjectDto extends Dto {
    id: number,
    semesterId: number,
    name: string,
    color: Color
}

File: backend/src/main/kotlin/outcobra/server/model/dto/manage/old/ManageDto.kt
package outcobra.server.model.dto.manage.old

import outcobra.server.model.dto.ColorDto
import java.time.LocalDate

data class ManageDto(
        val institutions: List<InstitutionDto>
)

data class InstitutionDto(
        val id: Long,
        val name: String,
        val schoolClasses: List<SchoolClassDto>
)

data class SchoolClassDto(
        val id: Long,
        val normalizedName: String,
        val institutionId: Long,
        val schoolYears: List<SchoolYearDto>
)

data class SchoolYearDto(
        val id: Long,
        val name: String,
        val validFrom: LocalDate,
        val validTo: LocalDate,
        val schoolClassId: Long,
        val semesters: List<SemesterDto>
)

data class SemesterDto(
        val id: Long,
        val name: String,
        val validFrom: LocalDate,
        val validTo: LocalDate,
        val schoolYearId: Long,
        val subjects: List<SubjectDto>
)

data class SubjectDto(
        val id: Long,
        val name: String,
        val color: ColorDto,
        val semesterId: Long
)
